src/Algebra/Properties/CancellativeCommutativeSemiring/GCD.agda

```diff
@@ -0,0 +1,115 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of the Greatest Common Divisor in
+-- CancellativeCommutativeSemiring.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra using (CancellativeCommutativeSemiring)
+open import Data.Product using (_,_; proj₁; proj₂)
+open import Data.Sum.Base using (_⊎_)
+open import Relation.Binary using (Decidable)
+import Relation.Binary.Reasoning.Setoid as EqReasoning
+open import Relation.Nullary using (Dec; yes; no)
+
+module Algebra.Properties.CancellativeCommutativeSemiring.GCD
+  {a ℓ} (R : CancellativeCommutativeSemiring a ℓ)
+  (open CancellativeCommutativeSemiring R)
+  where
+
+open import Algebra.Properties.Semiring.Primality semiring using (Coprime)
+open import Algebra.Properties.Semiring.Divisibility semiring
+open EqReasoning setoid
+import Algebra.Properties.CommutativeSemigroup *-commutativeSemigroup as Of*CSemig
+open import Algebra.Properties.CommutativeSemigroup.Divisibility
+  *-commutativeSemigroup using (x∣xy; x∣y∧z∣x/y⇒xz∣y; x∣y⇒zx∣zy)
+open import Algebra.Properties.CancellativeCommutativeSemiring R using (xy∣x⇒y∣1)
+
+---------------------------------------------------------------------------
+-- Re-exporting definition of GCD and its properties in semiring
+
+open import Algebra.Properties.Semiring.GCD semiring public
+
+---------------------------------------------------------------------------
+-- Properties of GCD
+
+x≉0⊎y≉0⇒Coprime[x/gcd,y/gcd] : ∀ {x y d} → x ≉ 0# ⊎ y ≉ 0# →
+                                ((mkIsGCD (q₁ , _) (q₂ , _) _) : IsGCD x y d) →
+                                Coprime q₁ q₂
+x≉0⊎y≉0⇒Coprime[x/gcd,y/gcd] x≉0∨y≉0 gcd@(mkIsGCD d∣x d∣y greatest) x/d∣z y/d∣z =
+  xy∣x⇒y∣1 (x≉0∨y≉0⇒gcd≉0 gcd x≉0∨y≉0) (greatest
+    (x∣y∧z∣x/y⇒xz∣y d∣x x/d∣z)
+    (x∣y∧z∣x/y⇒xz∣y d∣y y/d∣z))
+
+------------------------------------------------------------------------------
+-- gcd-s for two division-equivalent pairs
+-- are division-equivalent
+
+GCD-unique : ∀ {x x' y y' d d'} → x ∣∣ x' → y ∣∣ y' →
+             IsGCD x y d → IsGCD x' y' d' → d ∣∣ d'
+GCD-unique (x∣x' , x'∣x) (y∣y' , y'∣y)
```

Comment:
This property does not require cancellation and therefore should not be in this file.

Reply:
Indeed. Thank you.
It is moved to Algebra.Properties.Semiring.GCD.